ouster_client/src/lidar_scan.cpp

```diff
@@ -621,18 +622,60 @@ bool ScanBatcher::operator()(const uint8_t* packet_buf, LidarScan& ls) {
             next_valid_m_id = m_id + 1;
         }
 
-        // old header API; will be removed in a future release
-        ls.header(m_id) = {ts, encoder, status};
-
         // write new header values
-        ls.timestamp()[m_id] = ts.count();
+        ls.timestamp()[m_id] = ts;
         ls.measurement_id()[m_id] = m_id;
         ls.status()[m_id] = status;
 
         impl::foreach_field(ls, parse_field_col(), m_id, pf, col_buf);
     }
 
     return false;
-}  // namespace ouster
+}
+
+std::string to_string(const Imu& imu) {
+    std::stringstream ss;
+    ss << "Imu: ";
+    ss << "linear_accel: [";
+    for (size_t i = 0; i < imu.linear_accel.size(); ++i) {
+        if (i > 0) ss << ", ";
+        ss << imu.linear_accel[i];
+    }
+    ss << "]";
+    ss << ", angular_vel = [";
+    for (size_t i = 0; i < imu.angular_vel.size(); ++i) {
+        if (i > 0) ss << ", ";
+        ss << imu.angular_vel[i];
+    }
+    ss << "]";
+    ss << ", ts: [";
+    std::array<std::string, 3> labels{"sys_ts", "accel_ts", "gyro_ts"};
+    for (size_t i = 0; i < imu.ts.size(); ++i) {
+        if (i > 0) ss << ", ";
+        // just in case imu vector is bigger than 3 (for what we have labels)
+        if (i < labels.size()) {
+            ss << labels[i] << " = ";
+        }
+        ss << imu.ts[i];
```

Comment:
[nit] since currently imu.ts size is hard-coded to 3 we don't need the additional check. In case new requirements emerges and imu.ts has dynamic size we can revise this piece of code. Otherwise, I would rather not have this check.